apps/video/js/view.js

```diff
@@ -206,6 +208,25 @@
     dom.timeSlider.addEventListener('keypress', handleSliderKeypress);
 
     ForwardRewindController.init(dom.player, dom.seekForward, dom.seekBackward);
+
+    // Workaround to fix issue reported in Bug 1214157
+    // We would like to pause video playback when interrupted by competing audio
+    // Audio Channel API throws interruptbegin event after play (Bug 1192748)
+    // If performance.now() - lastPlayTime is small we ignore interruptbegin
+    // and pause video playback only when interrupted by competing audio.
+    // This workaround should be removed with the fix of Bug 1206581
+    dom.player.addEventListener('play', function() {
+      lastPlayTime = window.performance.now();
+    });
+
+    dom.player.addEventListener('mozinterruptbegin', function() {
+      // Ignore mozinterruptbegin event if it immediately follows play event
+      if ((window.performance.now() - lastPlayTime) > 500) {
```

Comment:
Tried and this third alternative won't work for view.js where we are receiving two interrupt-begin event after play, first interrupt-begin will set bool variable to false and second interrupt-begin event will check for false and pause video which is not desired.